libs/sdk/tools/src/generators/update-dependencies/generator.ts

```diff
@@ -149,27 +154,39 @@ function updateTestingImplicitDependencies(tree: Tree): void {
       projectConfig.targets ??= {};
       projectConfig.targets['build'] = {
         command: `echo ' 🏗️  build ${projectConfig.name}'`,
-        dependsOn: [
-          {
-            projects: [projectConfig.name.replace('-testing', '')],
-            target: 'build',
-          },
-        ],
+        inputs: ['buildInputs'],
+        dependsOn: [],
       };
       if (importedProjects.length > 0) {
         projectConfig.implicitDependencies = [
           ...new Set(importedProjects),
         ].sort((a, b) => a.localeCompare(b));
       }
       updateProjectConfiguration(tree, projectConfig.name, projectConfig);
+      const parentProject = projects.get(parentProjectName);
+      if (parentProject) {
+        if (parentProject?.targets?.['build']) {
+          parentProject.targets['build'].dependsOn = [
+            '^build',
+            {
+              projects: [projectConfig.name],
+              target: 'build',
+            },
+          ];
+        }
+        updateProjectConfiguration(tree, parentProjectName, parentProject);
+      }
```

Comment:
Make any inputs for, e.g., `core-testing:build`, force `core:build`. Since this also overrides `nx.json` file `targetDefaults`, include `^build` to force dependencies to build first.

One gap here is `^build` for, e.g., `core-testing` does not trigger `core:build`. I think that's ok since we ship everything every release (i.e., it won't be skipped), and `core-testing:test` would still be triggered so any breaking changes should still be caught. It's not possible to include b/c `^build` for `core-testing` includes `core:build`, which is circular.